interpreter/cling/include/cling/Interpreter/Transaction.h

```diff
@@ -316,6 +322,14 @@ namespace cling {
       m_Opts = CO;
     }
 
+    clang::NamespaceDecl* getDefinitionNS() const { return m_DefinitionNS; }
+    void setDefinitionNS(clang::NamespaceDecl* NS) {
+      assert(!m_DefinitionNS && "Transaction has a __cling_N5xxx NS?");
+      m_DefinitionNS = NS;
+      // Ensure `NS` is unloaded from the AST on transaction rollback, e.g. '.undo X'
+      append(reinterpret_cast<clang::Decl*>(NS));
```

Comment:
Eeew, `reinterpret_cast`? Can we outline this function (it's not perf critical anyway), and rely on the source's `#include` of `NamespaceDecl` to do the conversion-to-base correctly, getting rid of the `reinterpret_cast`?

Reply:
Fixed in commit 69e3453a12.